libsolidity/ast/AST.h

```diff
@@ -851,6 +853,12 @@ class CallableDeclaration: public Declaration, public VariableScope
 	ASTPointer<OverrideSpecifier> const& overrides() const { return m_overrides; }
 	std::vector<ASTPointer<VariableDeclaration>> const& returnParameters() const { return m_returnParameters->parameters(); }
 	ParameterList const& parameterList() const { return *m_parameters; }
+	SourceLocation const  prototypeLocation  () const override {
+		SourceLocation const& source = location();
+		SourceLocation functionLocation = parameterList().location();
```

Comment:
This is not necessarily a function. Note that you placed it inside `Declaration`. This includes also variables, events, errors, imports, contracts, structs, user-defined value types.

What the variable really represents is the result of the function so it's fine to call it that. `prototypeLocation` would also be okay if the compiler does not complain about it shadowing the function itself.
```suggestion
		SourceLocation result = parameterList().location();
```